src/modules/simulation/gz_plugins/moving_platform_controller/MovingPlatformController.cpp

```diff
@@ -0,0 +1,324 @@
+	_link_entity = _model.LinkByName(ecm, link_name);
+
+	if (!_link_entity) {
+		throw std::runtime_error("MovingPlatformController::Configure: Link \"" + link_name + "\" was not found. "
+					 "Please ensure that your model contains the corresponding link.");
+	}
+
+	_link = gz::sim::Link(_link_entity);
+
+	// Needed to report linear & angular velocity
+	_link.EnableVelocityChecks(ecm, true);
+
+	_startup_timer = gz::common::Timer();
+	_startup_timer.Start();
+
+	// Get velocity and orientation setpoints from env vars
+	{
+		const double vel_forward = ReadEnvVar("PX4_GZ_PLATFORM_VEL", 1.0);
+		const double heading_deg = ReadEnvVar("PX4_GZ_PLATFORM_HEADING_DEG", 0.0);
+		const double heading = GZ_DTOR(heading_deg);
+
+		_orientation_sp = gz::math::Quaterniond(0., 0., heading);
+
+		// Set initial orientation to match the heading
+		const auto optional_original_pose = _link.WorldPose(ecm);
+
+		if (optional_original_pose.has_value()) {
+			const gz::math::Pose3d original_pose = optional_original_pose.value();
+			const gz::math::Pose3d new_pose(original_pose.Pos(), _orientation_sp);
+			_model.SetWorldPoseCmd(ecm, new_pose);
+
+		} else {
+			gzwarn << "Unable to get initial platform pose. Heading will not be set" << std::endl;
+		}
+
+		// Velocity setpoint in world frame.
+		const gz::math::Vector3d _body_velocity_sp(vel_forward, 0., 0.);
+		_velocity_sp = _orientation_sp * _body_velocity_sp;
+	}
+
+	// Get gravity, model mass, platform height.
+	{
+		const auto world_entity = gz::sim::worldEntity(ecm);
+		const auto world = gz::sim::World(world_entity);
+		const auto gravity = world.Gravity(ecm);
+
+		if (gravity.has_value()) {
+			_gravity = gravity.value().Z();
+
+		} else {
+			gzwarn << "Unable to get gazebo world gravity. Keeping default of " << _gravity << std::endl;
+		}
+
+		auto inertial_component = ecm.Component<gz::sim::components::Inertial>(_link_entity);
+
+		if (inertial_component) {
+			_platform_mass = inertial_component->Data().MassMatrix().Mass();
+			_platform_diag_moments = inertial_component->Data().MassMatrix().DiagonalMoments();
+
+		} else {
+			gzwarn << "Unable to get inertial component for link " << link_name << ". Keeping default mass of " << _platform_mass <<
+			       std::endl;
+		}
+
+		auto pose_component = ecm.Component<gz::sim::components::Pose>(_link_entity);
+
+		if (pose_component) {
+			_platform_height_setpoint = pose_component->Data().Z();
+
+		} else {
+			gzwarn << "Unable to get pose component for link " << link_name << ". Keeping default platform height setpoint of " <<
+			       _platform_height_setpoint << std::endl;
+		}
+	}
+
+}
+
+void MovingPlatformController::PreUpdate(const gz::sim::UpdateInfo &_info, gz::sim::EntityComponentManager &ecm)
+{
+	getPlatformState(ecm);
+
+	const double dt_sec = std::chrono::duration<double>(_info.dt).count();
+	updateNoise(dt_sec);
+
+	// Wait for model to spawn, takes quite some time now...
+	const bool keep_stationary = _startup_timer.ElapsedTime() < 5s;
```

Comment:
Can you quickly explain why you need to keep it stationary for 5s?

Reply:
because the model takes 5s to spawn. If we don't wait for it, the platform moves away before it spawns.

Reply:
@mbjd Doing this based on time would not be reliable, as this would largely depend on the system. You can check if the entity exists through the gz API.

Reply:
@Jaeyoung-Lim true that's a better solution, done in latest commit.